### ISSUE TITLE
Using the pretrained model on test data

### ISSUE THREAD

**Issue by maryamkqamar** [XinhaoMei/ACT#4]
Hi thanks for the code. I want to use the pretrained model for making inference on my own data. So the data is only audio files (and no csv files for training). I set up the environment, then put data files in data folder (waveforms). Updated settings.yaml file as instructed (eval mode, pretrained model etc.). Could you please guide me about making inference now? (I tried running train.py but it gives errors about missing train.h5 file etc.). Thanks.

**Comment by XinhaoMei** [comment]
Hi, thanks for your interests.

To use your own dataset, you need to implement your own dataloader. And then you don't need to do evaluation as you don't have ground truth.

**Comment by maryamkqamar** [comment]
Thank you for the response.

**Comment by DthdZK** [comment]
So what should I do in detail or run which .py to inference on the test set of audiocaps? @XinhaoMei @maryamkqamar Thanks!

**Comment by XinhaoMei** [comment]
> So what should I do in detail or run which .py to inference on the test set of audiocaps? @XinhaoMei @maryamkqamar Thanks!

Hello,

We didn't provide a separete inference script in this repository. In the train.py, we do validation and test. You could try to adapt the test code in train.py for separate inference.

**Comment by DthdZK** [comment]
> > So what should I do in detail or run which .py to inference on the test set of audiocaps? @XinhaoMei @maryamkqamar Thanks!
> 
> Hello,
> 
> We didn't provide a separete inference script in this repository. In the train.py, we do validation and test. You could try to adapt the test code in train.py for separate inference.

Thank you so much！I will  adapt the test code in train.py for separate inference.
